### DISCUSSION TITLE
Dashboard performance on Docker/Synology

### DISCUSSION THREAD

**Discussion by Adelscott** [alexjustesen/speedtest-tracker#475] [Q&A]
It's running on Docker and the UI is extremely slow, changing page or clicking a button or a list is a pain.
No issues with the other dockers (Portainer, Radarr, etc ...) running on same host DS713+
Is there any tuning I could do ?

![image](https://user-images.githubusercontent.com/10946454/224551279-0a31eb75-880f-4d9c-8b03-3fa15f0f57b5.png)

**Comment by alexjustesen** [comment]
Couple of items to checkout...

1. Make sure your `PUID` and `PGID` are set correctly, if they aren't the internal view caching might not work correctly which can affect performance.
2. Additionally queries and other actions are cached in the database by default so I'd also make sure your DB is performant.
3. The app does "polling" in the background which will automatically update certain views (like the dashboard) every 5s, there is no downside to this approach but if you're used to SPA's then it might "feel" like a delay.

**Comment by Adelscott** [comment]
I think PUID and PGID are OK, I created a user like I did for the other containers:
![image](https://user-images.githubusercontent.com/10946454/224551564-abfec3e1-f4fb-444b-b245-5d266d8a2f74.png)

**Reply by alexjustesen** [reply]
`-AUTH=true` flag you don't need and can remove that, also make sure you have the latest image `v0.11.8` and reboot the container if you haven't already too.

**Comment by Adelscott** [comment]
Done, but no impact
![image](https://user-images.githubusercontent.com/10946454/224553074-4b157033-bbcf-42eb-8bc9-918227bcfd50.png)

**Reply by jeffmcm64** [reply]
Did you ever get this resolved, it's also pretty slow for me as well.  Takes 3 seconds to load anytime I click in the dashboard UI.  On the latest version, 0.14.3

I noticed while monitoring htop, a single CPU thread goes to 100% every time I'm browsing the dashboard.

**Reply by alexjustesen** [reply]
@jeffmcm64 in your environment do you have `CACHE_DRIVER` set as `file` or `database`?

**Reply by jeffmcm64** [reply]
@alexjustesen I don't appear to have a CACHE_DRIVER variable in my ENV file, unless I'm looking in the wrong spot.  I'm using Portainer to manage Docker and looking at the speedtest env settings.

**Reply by jeffmcm64** [reply]
@alexjustesen I went and added `CACHE_DRIVER` set as `file` and recreated the container.  It might be a little faster, it takes 2 seconds to load each page when navigating, except the main dashboard which takes 6 seconds to load.

**Reply by alexjustesen** [reply]
Are you using a sqlite DB or external DB? I've been tracking this application performance and trying to narrow down the issue. 

Only appears in production deployments, local dev is less than 1s.

**Reply by jeffmcm64** [reply]
@alexjustesen SQL DB, I spun it up using your installation template for Docker: https://docs.speedtest-tracker.dev/getting-started/installation/installation

I didn't edit or change much.

If it helps, I am running ESXi 8.0, which is hosting a Ubuntu VM, which is hosting Docker, managed by Portainer.  The VM has 4GB vCPU and 4GB vRAM allocated to it, and its doing very little, it hosts a pihole as well but that's all it does.

**Reply by alexjustesen** [reply]
Thanks for the details, this should help me track down the issue. 

I'm also seeing it on Synology, unRAID and rpi installs so it's not likely related to one host.

**Reply by jeffmcm64** [reply]
Feel free to let me know if you'd like further details or have me test anything.

I just spun it up in the last week, so I have no issues redoing it for testing.

I ran into this script from elsewhere, and they had a script which I compared with your template, and this is what I ran:

> version: "3.3" # Specifies the version of the Docker Compose file format
> 
> services: # Defines the services that make up the application
>   # Service: speedtest-tracker
>   speedtest-tracker:
>     container_name: speedtest-tracker # Name of the container, useful for referencing and managing the container
>     ports:
>       - "8080:80" # Maps port 8080 on the host to port 80 in the container; used for HTTP traffic
>       - "8443:443" # Maps port 8443 on the host to port 443 in the container; used for HTTPS traffic
>     environment: # Environment variables for the container, used to configure the container's software
>       - PUID=1000 # User ID for file permissions
>       - PGID=1000 # Group ID for file permissions
>       - DB_CONNECTION=mysql # Database connection type, indicating MySQL is used
>       - DB_HOST=db # Hostname of the database, linked to the db service
>       - DB_PORT=3306 # Port for the MySQL database
>       - DB_DATABASE=speedtest_tracker # Database name to be used
>       - DB_USERNAME=speedy # Database username
>       - DB_PASSWORD=password # Database password
>       - TZ=America/Edmonton # Timezone setting for the container; affects how time is reported
>     volumes: # Mounts host paths into the container, for persistent storage or sharing data
>       - "/etc/localtime:/etc/localtime:ro" # Mounts the host's localtime as read-only for accurate time syncing
>       - "/DATA/AppData/speedtest-tracker-2/config:/config" # Volume for storing configuration files
>       - "/DATA/AppData/speedtest-tracker-2/web:/etc/ssl/web" # Volume for storing SSL certificates or web-related files
>     image: "ghcr.io/alexjustesen/speedtest-tracker:latest" # Docker image to use, pulling the latest version
>     restart: unless-stopped # Container restart policy: restart if the container stops unexpectedly
>     depends_on: # Dependency setting, indicating that this service depends on the db service
>       - db # Ensures db service is started before speedtest-tracker
> 
>   # Service: db (Database)
>   db:
>     image: mariadb:10 # MariaDB image, version 10, used as the database server
>     restart: always # Restart policy: always restarts the container if it stops for any reason
>     environment: # Environment variables to configure MariaDB
>       - MARIADB_DATABASE=speedtest_tracker # Database to be created automatically
>       - MARIADB_USER=speedy # Username for the database, used by speedtest-tracker
>       - MARIADB_PASSWORD=password # Password for the database user
>       - MARIADB_RANDOM_ROOT_PASSWORD=true # Enables random password generation for the root user for security
>     volumes: # Mount for database storage, ensuring data persists across container restarts and removal
>       - /DATA/AppData/speedtest-tracker-2/db:/var/lib/mysql # Maps the host directory to the container's database storage location
